### PR TITLE
Fix/get events loop condition

### LINKED ISSUE #233: Bug: Inverted loop condition in GetEvents causes immediate termination for bounded range queries
## Description

A bug was discovered in the `GetEvents` method in `internal/persistence/provider/bbolt/store.go` around lines 96-97. The loop end condition is inverted, causing range retrieval to fail when `eventIndexEnd != 0`.

## Problem

The current condition uses:
```
bytes.Compare(k, conv.Itob(uint64(eventIndexEnd))) > 0
```

This checks `k > end`, which causes the loop to terminate immediately for the common case where `start <= end`. The loop should continue while `k <= end` (or unbounded when `end == 0`).

## Impact

This breaks bounded event range retrieval functionality, affecting any code that tries to fetch events within a specific range.

## Location

File: `internal/persistence/provider/bbolt/store.go`
Lines: ~96-97 (in the `GetEvents` method)

## Discovery Context

This issue was discovered during code review of PR #232.

**References:**
- PR: https://github.com/ccamel/playground-protoactor.go/pull/232
- Comment: https://github.com/ccamel/playground-protoactor.go/pull/232#discussion_r2288895060
- Discovered by: @coderabbitai
- Reported by: @ccamel

### PR DESCRIPTION
**Pull request by ccamel**
Fixes #233 (+ table-driven tests to cover this).

## Summary by CodeRabbit

- New Features
  - Event retrieval now supports both bounded and unbounded ranges for flexible streaming.

- Refactor
  - Streamlined iteration and early-exit handling for missing actor data.
  - Clearer error reporting and logging for missing or malformed event payloads.
  - Minor signature formatting change with no behavioral impact.

- Tests
  - Expanded coverage: unbounded streaming from any start and table-driven bounded-range scenarios, including edge cases.

- Chores
  - Commit validation updated to allow an additional emoji type.

### CODE REVIEW

File: internal/persistence/provider/bbolt/store.go [hunk 1]
```diff
@@ -82,32 +82,35 @@ func (s *Store) PersistSnapshot(actorName string, record *persistencev1.Snapshot
...
-func (s *Store) GetEvents(actorName string, eventIndexStart int, eventIndexEnd int, callback func(e *persistencev1.EventRecord)) {
-	err := s.db.View(func(tx *bolt.Tx) error {
-		actorBucket := s.
-			eventsBucket(tx).
-			Bucket([]byte(actorName))
+func (s *Store) GetEvents(actorName string, eventIndexStart, eventIndexEnd int, callback func(e *persistencev1.EventRecord)) {
+	startKey := conv.Itob(uint64(eventIndexStart))
+	endKey := conv.Itob(uint64(eventIndexEnd))
+	unbounded := eventIndexEnd == 0
+
```
[the remark this answers is not in the document]
Comment (ccamel) [reply internal/persistence/provider/bbolt/store.go:89]: Good point mate. By design ranges are `uint`, so _make illegal states unrepresentable_ would normally apply… But since the contract forces `int`, we can't enforce that at the type level. Clamping negatives feels like the right defensive move.

**Comment by ccamel** [comment]
@coderabbitai resolve
